R/sample.R

```diff
@@ -68,7 +72,16 @@ mcstate_sample <- function(model, sampler, n_steps, initial = NULL) {
   colnames(history_pars) <- model$parameters
 
   ## I'm not sure about the best name for this
-  details <- sampler$finalise(state, model)
+  details <- sampler$finalise(state, model, rng)
+
+  if (!identical(get_r_rng_state(), r_rng_state)) {
+    cli::cli_warn(c(
+      "Detected use of R's random number generators",
+      i = paste("Your model has used R's random number generators (e.g.,",
```

Comment:
```suggestion
      i = paste("Your model has used R's random number generators (e.g.",
```

Reply:
https://en.wiktionary.org/wiki/e.g. - e.g. is usually followed by a comma :)

Reply:
damn, i had no idea :open_mouth: